### PR TITLE
Reputation contract improvements

### LINKED ISSUE #64: Reputation contract improvements
External method isActive is declared in Reputation interface. However this method is not implemented in Reputation contract.

**Proposal 1**
<strike>
There is already implementation of `isActive modifier` in reputation contract. Proposal is to implement `Reputation.isActiveValidator` method. `isActive modifier` should call  isActiveValidator method.
**Method Signature**
```
function isActiveValidator(address _validator)
        public
        view
        returns(bool)
    {}
```
 </strike>

**Proposal 2**

- Introduce new interface `isSlashed` in reputation contract with below function signature:
```
function isSlashed(address _validator)
        public
        view
        returns(bool)
    {}
```
Remove interface isActive from Reputation contract. Keep the isActive modifier as it is.
Core.registerVote should call Reputation.isSlashed method.
Consensus.enterCommittee should call Reputation.isSlashed method.

- Even after user gets logged out in Reputation contract, he is accountable for his vote. 
   Consensus.logout is different from Reputation.logout. 
   - Rename `Reputation.logout` to `Reputation.deRegister`. 
   - Update status `LoggedOut` to `DeRegistered`. 

- Rename method `Reputation.join` to `Reputation.stake`

### PR DESCRIPTION
**Pull request by abhayks1**
- Introduce new interface `isSlashed` in reputation contract with below function signature:
```
function isSlashed(address _validator)
        public
        view
        returns(bool)
    {}
```
Remove interface isActive from Reputation contract. Keep the isActive modifier as it is.
Core.registerVote should call this method.
Consensus.enterCommittee should call this method.

- Even after user gets logged out in Reputation contract, he is accountable for his vote. 
   - Rename `Reputation.logout` to `Reputation.deregister` 
   - Update status `LoggedOut` to `DeRegistered` 

- Rename method `Reputation.join` to `Reputation.stake`

Fixes #64

### CODE REVIEW

File: contracts/reputation/Reputation.sol [hunk 1]
```diff
@@ -48,10 +48,10 @@ contract Reputation is ConsensusModule {
         /** Validator has put up stake and participates in consensus */
         Staked,
 
-        /** Validator has logged out and no longer participates in consensus */
-        LoggedOut,
+        /** Validator has deregistered and no longer participates in consensus */
+        DeRegistered,
```
Comment (benjaminbollen) [inline contracts/reputation/Reputation.sol]: Mind capitalisation

File: contracts/reputation/Reputation.sol [hunk 2]
```diff
@@ -130,11 +130,11 @@ contract Reputation is ConsensusModule {
         _;
     }
```
Comment (benjaminbollen) [inline contracts/reputation/Reputation.sol:133]: Similar capitlisation

Comment (abhayks1) [reply contracts/reputation/Reputation.sol:133]: Done!

File: contracts/reputation/Reputation.sol (hunk 1 above)
Comment (abhayks1) [reply contracts/reputation/Reputation.sol]: Changes done!

**Comment by abhayks1** [comment]
PR ready for review again 📗

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Pull request merged**